File: app/src/main/java/com/mrhuang/demo/chapter1/SurfaceViewActivity.java
package com.mrhuang.demo.chapter1;

import android.os.Bundle;
import android.view.SurfaceView;

import com.mrhuang.demo.BaseActivity;
import com.mrhuang.demo.R;

public class SurfaceViewActivity extends BaseActivity {

    SurfaceView surfaceView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_surface_view);
    }


    @Override
    protected void getViews() {
        surfaceView = findViewById(R.id.surfaceView);
    }

    @Override
    protected void viewCreated() {
    }
}

File: app/src/main/java/com/mrhuang/demo/chapter1/SurfaceViewActivity.kt
package com.mrhuang.demo.chapter1

import android.os.Bundle
import android.view.SurfaceView
import com.mrhuang.demo.BaseActivity
import com.mrhuang.demo.R

class SurfaceViewActivity : BaseActivity() {
    var surfaceView: SurfaceView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_surface_view)
    }

    override fun getViews() {
        surfaceView = findViewById(R.id.surfaceView)
    }

    override fun viewCreated() {}
}